packages/qdrant-loader/src/qdrant_loader/core/state/transitions.py

```
@@ -213,7 +213,25 @@ async def update_document_states_batch(
             except Exception as e:  # noqa: BLE001 - reported to caller, not swallowed
                 results.append((document, None, e))
 
-        await session.commit()
+        try:
+            await session.commit()
+        except Exception as commit_error:  # noqa: BLE001 - return as per-document failures
+            try:
+                await session.rollback()
+            except Exception:
+                # Best-effort rollback; keep the original commit error as the reported cause.
+                pass
```

Ruff (0.15.21)

[warning] Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

(SIM105)